saw-core/src/Verifier/SAW/Term/Pretty.hs

```diff
@@ -664,8 +690,9 @@ ppLets global_p ((idx, (t_rhs,_)):idxs) bindings baseDoc =
   do isBound <- isJust <$> memoLookupM idx
      if isBound then ppLets global_p idxs bindings baseDoc else
        do doc_rhs <- ppTerm' PrecTerm t_rhs
-          withMemoVar global_p idx $ \memo_var ->
-            ppLets global_p idxs ((memo_var, doc_rhs):bindings) baseDoc
+          withMemoVar global_p idx $ \case
+            Just memo_var -> ppLets global_p idxs ((memo_var, doc_rhs):bindings) baseDoc
+            Nothing -> ppLets global_p idxs bindings baseDoc
```

Comment:
These cases are nearly identical, aside from what bindings are passed it. I'd prefer to make this obvious by writing it more like this:

```hs
          withMemoVar global_p idx $ \mbMemoVar ->
            let bindings' = case mbMemoVar of
                                   Just memo_var -> (memo_var, doc_rhs):bindings
                                   Nothing -> bindings in
            ppLets global_p idxs bindings' baseDoc
```

Reply:
Fair enough, I shall fix it.